test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs

```diff
@@ -496,4 +559,181 @@ private async Task DownloadFileAsync(Uri url, string filePath)
+        }
+        else if (packageType == PackageType.Rpm)
+        {
+            return GetRpmPackageDependencies(packagePath);
+        }
+
+        return [];
+    }
+
+    private List<string> GetRpmPackageDependencies(string packagePath)
+    {
+        throw new NotImplementedException("https://github.com/dotnet/arcade/pull/16079 is required for getting RPM package dependencies.");
+    }
+
+    private List<string> GetDebianPackageDependencies(string packagePath)
+    {
+        try
+        {
+            using FileStream debStream = File.OpenRead(packagePath);
+            using ArReader ar = new ArReader(debStream, false);
+
+            // Find control.tar.* entry
+            while (ar.GetNextEntry() is ArEntry arEentry)
+            {
+                if (!arEentry.Name.StartsWith("control.tar", StringComparison.Ordinal))
+                {
+                    continue;
+                }
+
+                using MemoryStream controlArchiveData = new MemoryStream((int)arEentry.DataStream.Length);
+                arEentry.DataStream?.CopyTo(controlArchiveData);
+                controlArchiveData.Position = 0;
+
+                Stream decompressed =
+                    arEentry.Name.EndsWith(".gz", StringComparison.OrdinalIgnoreCase) ?
+                        new GZipStream(controlArchiveData, CompressionMode.Decompress, leaveOpen: false) :
+                    // Future compressors can be added here (xz, zst). Fallback: treat as plain.
+                        controlArchiveData;
+
+                using (decompressed)
+                {
+                    // Read tar entries to find "control" file
+                    using TarReader tarReader = new TarReader(decompressed, leaveOpen: false);
+                    TarEntry? entry;
+                    while ((entry = tarReader.GetNextEntry()) is not null)
+                    {
+                        if (entry.Name
+                            .TrimStart('.', '/')
+                            .Equals("control", StringComparison.Ordinal))
+                        {
+                            using MemoryStream controlFileData = new MemoryStream();
+                            entry.DataStream?.CopyTo(controlFileData);
+                            string controlContent = Encoding.UTF8.GetString(controlFileData.ToArray());
+                            File.WriteAllText(Path.Combine(Path.GetTempPath(), "control.txt"), controlContent);
+                            return ParseDebControlDependencies(controlContent);
+                        }
+                    }
+                }
+            }
+
+            _outputHelper.WriteLine($"No control.tar.* entry found in {packagePath}");
+            return [];
+        }
+        catch (Exception ex)
+        {
+            _outputHelper.WriteLine($"Error parsing DEB package '{packagePath}': {ex}");
+            return [];
+        }
+    }
+
+    private static List<string> ParseDebControlDependencies(string contents)
```

Comment:
Fixed indentation of this method with https://github.com/dotnet/dotnet/pull/2221/commits/4ed8bb06b40117a6ea51f4bb4de7b9dfdff360f0